faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -920,7 +920,131 @@ class ClientSpec
       // Identify
       val identifyR = client.query(Identify(createR(RefField), "abcdefg")).futureValue
       identifyR.to[Boolean].get shouldBe true
-      }
+   }
+
+  it should "create_access_provider successful creation" in {
+    val roleName = "my_role" + aRandomString
+    val providerName = "my_provider" + aRandomString
+    val issuerName = "my_issuer" + aRandomString
+    val fullUri = s"https://${aRandomString}.auth0.com"
+
+    val roleV = adminClient.query(CreateRole(Obj(
+      "name" -> roleName,
+      "privileges" -> Obj(
+        "resource" -> Databases(),
+        "actions" -> Obj("read" -> true)
+      )
+    ))).futureValue
+
+    val providerV = adminClient.query(CreateAccessProvider(Obj(
+      "name" -> providerName,
+      "issuer" -> issuerName,
+      "jwks_uri" -> fullUri,
+      "roles" -> ArrayV(roleV(RefField))))
+    ).futureValue
+
+    providerV("roles").to[Seq[RefV]].get.size shouldBe 1
```

Comment:
Could we check for the rest of the fields here as well (e.g, `name`, `issuer`, etc)? :)

There's also a _dynamic_ field called `audience` which is added to the `AccessProvider` upon creation. Since its value would be a bit more tricky to determine before creation, we could just check if the value is present in the response. 

Something along the lines:

```scala
providerV("audience").toOpt shouldBe defined
```